File: backend/src/main/java/com/hgw/gestionale/color/mapper/ColorMapper.java
package com.hgw.gestionale.color.mapper;

import com.hgw.gestionale.color.dto.ColorResponse;
import com.hgw.gestionale.color.dto.CreateColorRequest;
import com.hgw.gestionale.color.dto.UpdateColorRequest;
import com.hgw.gestionale.color.entity.Color;

public final class ColorMapper {
    private ColorMapper() {}

    public static Color toEntity(CreateColorRequest request) {
        return Color.builder()
                .nome(request.nome())
                .build();
    }

    public static ColorResponse toResponse(Color color) {
        return new ColorResponse(
                color.getId(),
                color.getNome()
        );
    }

    public static void updateEntity(Color color, UpdateColorRequest request) {
        color.setNome(request.nome());
    }
}

File: backend-node/src/mapper/color.mapper.ts
import { ColorResponse } from "../dto/color.dto";
import { colors } from "../schema/colors";

type ColorEntity = typeof colors.$inferSelect;

export class ColorMapper {

    static toResponse(entity: ColorEntity): ColorResponse {
        return {
            id: entity.id,
            nome: entity.nome,
        };
    }

}
